tests/lang/s02_core/s02_procedures/t02_argument_passing_mutate_bug.nim

```diff
@@ -0,0 +1,39 @@
+discard """
+description: '''
+Mutating variable while passing it to the procedure
+arguments works differently on various GCs.
+'''
+knownIssue: "https://github.com/nim-lang/Nim/issues/18099"
```

Comment:
this is also broken for expressions, yes?

Reply:
Yes? I'm not sure what do you mean though - `doAssert impl(i, (inc i; i), (inc i; i)) == (0, 1, 2)` is tested (L24-28), and `(inc i; i)` is an expression.

Reply:
I meant just the bare expression, see below, but I think maybe that's been caught already in the expression spec and I just missed it.

```nim
let i = 0
doAssert 3 == (inc i; i) + (inc i; i)
```